src/writer/update-proposal.ts

```diff
@@ -0,0 +1,186 @@
+  const isAuthorized = isAddressAuthorized(address, space);
+
+  if (onlyAuthors && !isAuthorized) return 'only space authors can propose';
+  if (!isAuthorized) {
+    try {
+      const validationName = space.validation?.name || 'basic';
+      const validationParams = space.validation?.params || {};
+      const minScore = space.validation?.params?.minScore || space.filters?.minScore;
+
+      let isValid = false;
+      // default case
+      if (validationName === 'any' || (validationName === 'basic' && !minScore)) {
+        isValid = true;
+      } else {
+        if (validationName === 'basic') {
+          validationParams.minScore = minScore;
+          validationParams.strategies = space.validation?.params?.strategies || space.strategies;
+        }
+
+        isValid = await snapshot.utils.validate(
+          validationName,
+          address,
+          space.id,
+          space.network,
+          'latest',
+          validationParams,
+          { url: scoreAPIUrl }
+        );
+      }
+
+      if (!isValid) return 'validation failed';
+    } catch (e) {
+      capture(e, { contexts: { input: { space: space.id, address } } });
+      log.warn(
+        `[writer] Failed to check proposal validation, ${space.id}, ${address}, ${JSON.stringify(
+          e
+        )}`
+      );
+      return 'failed to check validation';
+    }
+  }
+}
+
+export async function verify(body): Promise<any> {
+  const msg = jsonParse(body.msg);
+
+  const schemaIsValid = snapshot.utils.validateSchema(snapshot.schemas.updateProposal, msg.payload);
+  if (!schemaIsValid) {
+    log.warn('[writer] Wrong proposal format', schemaIsValid);
+    return Promise.reject('wrong proposal format');
+  }
+
+  if (!isChoicesValid(msg.payload)) {
+    return Promise.reject(`wrong choices for ${msg.payload.type} type voting`);
+  }
+
+  const proposal = await getProposal(msg.space, msg.payload.proposal);
+  if (!proposal) return Promise.reject('unknown proposal');
+
+  if (proposal.author !== body.address) return Promise.reject('Not the author');
+
+  const space = await getSpace(msg.space);
+  space.id = msg.space;
+
+  const spaceUpdateError = getSpaceUpdateError(msg.payload, space);
+  if (spaceUpdateError) return Promise.reject(spaceUpdateError);
+
+  if (isScamDetected(body.address, msg.payload))
+    return Promise.reject('scam proposal detected, contact support');
+
+  const authorizationError = await checkAuthorization(body.address, space);
+  if (authorizationError) return Promise.reject(authorizationError);
+
+  return Promise.resolve(proposal);
+}
+
+export async function action(body, ipfs): Promise<void> {
+  const msg = jsonParse(body.msg);
+  const space = msg.space;
+
+  const author = getAddress(body.address);
+  const updated = parseInt(msg.timestamp);
+  const metadata = msg.payload.metadata || {};
+  const plugins = JSON.stringify(metadata.plugins || {});
+
+  const proposal = {
+    ipfs,
+    author,
+    updated,
+    space,
```

Comment:
Isn't this always the same value before and after edit ? Maybe it can be ignored